src/vm/threads.h

```diff
@@ -2907,15 +2907,34 @@ class Thread: public IUnknown
     DWORD Join(DWORD timeout, BOOL alertable);
     DWORD JoinEx(DWORD timeout, WaitMode mode);
 
+    struct EXTENDED_CONTEXT
```

Comment:
If you derived the EXTENDED_CONTEXT from CONTEXT instead, you could get rid of all the casts to CONTEXT* that you had to add and it seems to me it would be cleaner that way. We are extending the CONTEXT, so the "is a" relation makes sense.